server/start.go

```diff
@@ -640,3 +611,322 @@ func startApp(svrCtx *Context, appCreator types.AppCreator, opts StartCmdOptions
 	}
 	return app, cleanupFn, nil
 }
+
+// InPlaceTestnetCreator utilizes the provided chainID and operatorAddress as well as the local private validator key to
+// control the network represented in the data folder. This is useful to create testnets nearly identical to your
+// mainnet environment.
+func InPlaceTestnetCreator(testnetAppCreator types.AppCreator) *cobra.Command {
+	opts := StartCmdOptions{}
+	if opts.DBOpener == nil {
+		opts.DBOpener = OpenDB
+	}
+
+	cmd := &cobra.Command{
+		Use:   "in-place-testnet [newChainID] [newOperatorAddress]",
+		Short: "Create and start a testnet from current local state",
+		Long: `Create and start a testnet from current local state.
+After utilizing this command the network will start. If the network is stopped,
+the normal "start" command should be used. Re-using this command on state that
+has already been modified by this command could result in unexpected behavior.
+
+Additionally, the first block may take up to one minute to be committed, depending
+on how old the block is. For instance, if a snapshot was taken weeks ago and we want
+to turn this into a testnet, it is possible lots of pending state needs to be committed
+(expiring locks, etc.). It is recommended that you should wait for this block to be committed
+before stopping the daemon.
+`,
+		Example: "in-place-testnet localosmosis osmo12smx2wdlyttvyzvzg54y2vnqwq2qjateuf7thj",
+		Args:    cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			serverCtx := GetServerContextFromCmd(cmd)
+			_, err := GetPruningOptionsFromFlags(serverCtx.Viper)
+			if err != nil {
+				return err
+			}
+
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			withCMT, _ := cmd.Flags().GetBool(flagWithComet)
+			if !withCMT {
+				serverCtx.Logger.Info("starting ABCI without CometBFT")
+			}
+
+			newChainID := args[0]
+			newOperatorAddress := args[1]
+
+			// Confirmation prompt to prevent accidental modification of state.
+			reader := bufio.NewReader(os.Stdin)
+			fmt.Println("This operation will modify state in your data folder and cannot be undone. Do you want to continue? (y/n)")
+			text, _ := reader.ReadString('\n')
+			response := strings.TrimSpace(strings.ToLower(text))
+			if response != "y" && response != "yes" {
+				fmt.Println("Operation canceled.")
+				return nil
+			}
+
+			// Set testnet keys to be used by the application.
+			// This is done to prevent changes to existing start API.
+			serverCtx.Viper.Set(KeyIsTestnet, true)
+			serverCtx.Viper.Set(KeyNewChainID, newChainID)
+			serverCtx.Viper.Set(KeyNewOpAddr, newOperatorAddress)
```

Comment:
We set keys in Viper in order to prevent changing existing APIs. This allows us to share the same flow across `start` and `in-place-testnet` commands.